lib/spack/docs/developer_guide.rst

```diff
@@ -802,46 +802,9 @@ Publishing a release on GitHub
 Updating `develop` and `releases/latest`
```

Comment:
Should the order in this title change, just to match the order in the section?  My OCD is kicking in.

Reply:
I thought about it and decided it wasn't worth it. I'll do it now.